src/vm/gcenv.os.cpp

```diff
@@ -30,6 +30,47 @@
 uint32_t g_pageSizeUnixInl = 0;
 #endif
 
+#if !defined(FEATURE_PAL)
+
+static bool g_SeLockMemoryPrivilegeAcquired = false;
+
+bool InitLargePagesPrivilege()
+{
+    TOKEN_PRIVILEGES tp;
+    LUID luid;
+    if (!LookupPrivilegeValueW(nullptr, SE_LOCK_MEMORY_NAME, &luid))
+    {
+        return false;
+    }
+
+    tp.PrivilegeCount = 1;
+    tp.Privileges[0].Luid = luid;
+    tp.Privileges[0].Attributes = SE_PRIVILEGE_ENABLED;
+
+    HANDLE token;
+    if (!OpenProcessToken(::GetCurrentProcess(), TOKEN_ADJUST_PRIVILEGES, &token))
+    {
+        return false;
+    }
+
+    auto retVal = AdjustTokenPrivileges(token, FALSE, &tp, 0, nullptr, 0);
```

Comment:
See the previous comment on using `auto`

Reply:
thanks for noticing this, Jan. clearly I did not review this code careful enough. I don't think using auto is needed at all; and I don't think we need to call GLE 'cause AdjustTokenPrivileges already returns success or fail. GLE is just to get extended error info. 

---
In reply to: [271181058](https://github.com/dotnet/coreclr/pull/23251#discussion_r271181058) [](ancestors = 271181058)

Reply:
@Maoni0 GetLastError() is required. AdjustTokenPrivileges will succeed but the GetLastError also has to be zero otherwise it can return things like 1310 (ERROR_NOT_ALL_ADJUSTED) and I'm not sure if there are others as well.

Reply:
I see. that's the first win32 API I've ever seen where GLE is actually required to determine whether the API has succeeded.

Reply:
There are number of APIs like that actually...

Reply:
apparently not the ones I normally use :)